### PR TITLE
[fix](nereids)fix bug of exists subquery with limit clause

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

create table t1(c1 int, c2 int);
create table t2(c1 int, c2 int);
insert into t1 values (1,1);
insert into t2 values (1,1);

select * from t1 where exists (select * from t2 where t1.c1 = t2.c1 limit 0);

the result should be empty set.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java [hunk 1]
```diff
@@ -72,7 +73,13 @@ public Expression visitNot(Not not, CascadesContext context) {
     @Override
     public Expression visitExistsSubquery(Exists exists, CascadesContext context) {
         AnalyzedResult analyzedResult = analyzeSubquery(exists);
-
+        if (analyzedResult.rootIsLimitZero()) {
+            return BooleanLiteral.of(exists.isNot());
+        }
+        if (analyzedResult.isCorrelated() && analyzedResult.rootIsLimitWithOffset()) {
+            throw new AnalysisException("Unsupported correlated subquery with a LIMIT clause with offset > 0 "
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:80]: select * from t1 where not exists (select * from t2 where t1.c1 = t2.c1 limit 0 offset 1);
Means doris will throw exception ? I think it makes sence to return full set of t1.
and for exists, it returns empty sets also.

Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:80]: ignore above cases.
For this case, select * from t1 where [not] exists (select * from t2 where t1.c1 = t2.c1 limit 1 offset 1);
will doris throw exception?

**Review by xzj7019 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java (hunk 1 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java:80]: currently select * from t1 where [not] exists (select * from t2 where t1.c1 = t2.c1 limit 1 offset 1); will return wrong result.

**Pull request merged**